### PR TITLE
[FLINK-11563] Translate "Getting Help" page into Chinese

### PR DESCRIPTION
**Pull request by sunnyholyd**
Translate "Getting Help" page into Chinese

### CODE REVIEW

**Comment by XuQianJin-Stars** [comment]
Thanks for the effort.

Some formatting issues can refer to the Flink Translation Specifications:
https://docs.google.com/document/d/1zhGyPU4bVJ7mCIdUTsQr9lHLn4vltejhngG_SOV6g1w/edit#heading=h.by2zo7lwgzr5

**Comment by sunnyholyd** [comment]
> Hi, @sunnyholyd
> 
> Thanks for your effort. I left some comments but haven't finished the review yet. Could you refine the content refer to the Flink Translation Specifications as @XuQianJin-Stars posted ？
> 
> Best
> Kaibo



> Hi, @sunnyholyd
> 
> Thanks for your effort. I left some comments but haven't finished the review yet. Could you refine the content refer to the Flink Translation Specifications as @XuQianJin-Stars posted ？
> 
> Best
> Kaibo

Hi, @kaibozhou 
Thanks for your review. I will fix the issues involved in your comments，and refine the content refer to the Flink Translation Specifications.

Best
Sunnyd

File: gettinghelp.zh.md [hunk 1]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
-See [Type Extraction and Serialization]({{ site.docs-snapshot }}/dev/types_serialization.html) for
-an in-depth discussion of how Flink handles types.
+请参阅 [类型提取和序列化]({{ site.docs-snapshot }}/dev/types_serialization.html) 深入讨论 Flink 如何处理类型。
 
-### I see a ClassCastException: X cannot be cast to X.
+### 我看到一个 ClassCastException: X cannot be cast to X.
 
-When you see an exception in the style `com.foo.X` cannot be cast to `com.foo.X` (or cannot be assigned to `com.foo.X`), it means that
-multiple versions of the class `com.foo.X` have been loaded by different class loaders, and types of that class are attempted to be assigned to each other.
+当你看到 `com.foo.X` cannot be cast to `com.foo.X` ( 或者 cannot be assigned to `com.foo.X`) 样式的异常时，这意味着 `com.foo.X` 类的多个版本已经由不同的类加载器加载，并且尝试相互分配该类的类型。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 并且尝试相互分配该类的类型。
->
并且尝试相互赋值。

File: gettinghelp.zh.md [hunk 2]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
-The reason for that can be:
+原因可能是:
 
-  - Class duplication through `child-first` classloading. That is an intended mechanism to allow users to use different versions of the same
-    dependencies that Flink uses. However, if different copies of these classes move between Flink's core and the user application code, such an exception
-    can occur. To verify that this is the reason, try setting `classloader.resolve-order: parent-first` in the configuration.
-    If that makes the error disappear, please write to the mailing list to check if that may be a bug.
+  - 通过 `child-first` 的类加载方式实现类复制。这是一种预期的机制，该机制允许用户使用相同依赖的不同版本。然而，如果这些类的不同副本在 Flink 的核心代码和用户应用程序代码之间移动，则可能发生这种异常。为了验证这个原因，请尝试在配置中设置 `classloader.resolve-order：parent-first`。如果这可以使错误消失，请写信到邮件列表以检查是否可能是错误。
+  
+  - 从不同的执行中尝试缓存类，例如使用像 Guava 的 Interners 或 Avro 的 Schema 等通用工具进行缓存操作。尝试不使用 Interners，或减少 interner/cache 的使用范围，以确保每当新任务执行开始时都会创建新的缓存。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 当新任务执行开始时
->
当新任务开始执行时

File: gettinghelp.zh.md [hunk 3]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
+当你看到 `com.foo.X` cannot be cast to `com.foo.X` ( 或者 cannot be assigned to `com.foo.X`) 样式的异常时，这意味着 `com.foo.X` 类的多个版本已经由不同的类加载器加载，并且尝试相互分配该类的类型。
 
-The reason for that can be:
+原因可能是:
 
-  - Class duplication through `child-first` classloading. That is an intended mechanism to allow users to use different versions of the same
-    dependencies that Flink uses. However, if different copies of these classes move between Flink's core and the user application code, such an exception
-    can occur. To verify that this is the reason, try setting `classloader.resolve-order: parent-first` in the configuration.
-    If that makes the error disappear, please write to the mailing list to check if that may be a bug.
+  - 通过 `child-first` 的类加载方式实现类复制。这是一种预期的机制，该机制允许用户使用相同依赖的不同版本。然而，如果这些类的不同副本在 Flink 的核心代码和用户应用程序代码之间移动，则可能发生这种异常。为了验证这个原因，请尝试在配置中设置 `classloader.resolve-order：parent-first`。如果这可以使错误消失，请写信到邮件列表以检查是否可能是错误。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 请写信到邮件列表以检查是否可能是错误。
->
请写信到邮件列表以确认是否可能是bug。

File: gettinghelp.zh.md [hunk 4]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
+  
+  - 从不同的执行中尝试缓存类，例如使用像 Guava 的 Interners 或 Avro 的 Schema 等通用工具进行缓存操作。尝试不使用 Interners，或减少 interner/cache 的使用范围，以确保每当新任务执行开始时都会创建新的缓存。
 
-  - Caching of classes from different execution attempts, for example by utilities like Guava’s Interners, or Avro's Schema cache.
-    Try to not use interners, or reduce the scope of the interner/cache to make sure a new cache is created whenever a new task
-    execution is started.
+### 我有一个 AbstractMethodError 或 NoSuchFieldError 错误。
 
-### I have an AbstractMethodError or NoSuchFieldError.
+此类错误通常表示某些依赖版本的混淆。这意味着在执行期间加载了不同版本的依赖项（库），而不是编译代码的版本。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 此类错误通常表示某些依赖版本的混淆
->
此类错误通常表示混淆了某些依赖的版本。

File: gettinghelp.zh.md [hunk 5]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
 
-Such errors typically indicate a mix-up in some dependency version. That means a different version of a dependency (a library)
-is loaded during the execution compared to the version that code was compiled against.
+从 Flink 1.4.0 开始，在默认激活 `child-first` 类加载方式的情况下，你程序的 JAR 文件中的依赖更可能带有不同的版本，相比 Flink core 所使用的依赖或类路径中的其他依赖（例如来自 Hadoop ）而言。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 在默认激活 `child-first` 类加载方式的情况下，你程序的 JAR 文件中的依赖更可能带有不同的版本，相比 Flink core 所使用的依赖或类路径中的其他依赖（例如来自 Hadoop ）而言。

->

在默认激活 `child-first` 类加载方式的情况下，相比 Flink core 所使用的依赖或类路径中的其他依赖（例如来自 Hadoop ）而言，应用程序 JAR 文件中的依赖更可能带有不同的版本。

File: gettinghelp.zh.md [hunk 6]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
 
-From Flink 1.4.0 on, dependencies in your application JAR file may have different versions compared to dependencies used
-by Flink's core, or other dependencies in the classpath (for example from Hadoop). That requires `child-first` classloading
-to be activated, which is the default.
+如果你在 Flink 1.4 以上的版本中看到这些问题，则可能是出现了以下情况之一：
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 则可能是出现了以下情况之一：
->
则可能是属于以下某种情况：？

File: gettinghelp.zh.md [hunk 7]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
+  - 你的程序代码中存在依赖项版本冲突，确保所有依赖项版本都一致。
+  - 你与一个 Flink 不能支持 `child-first` 类加载的库发生了冲突。目前会产生这种情况的有 Scala 标准库类、Flink 自己的类、日志 API 和任何 Hadoop 核心类。
 
-If you see these problems in Flink 1.4+, one of the following may be true:
-  - You have a dependency version conflict within your application code. Make sure all your dependency versions are consistent.
-  - You are conflicting with a library that Flink cannot support via `child-first` classloading. Currently these are the
-    Scala standard library classes, as well as Flink's own classes, logging APIs, and any Hadoop core classes.
+### 即使事件正在发生，我的 DataStream 程序也没有输出。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 即使事件正在发生，我的 DataStream 程序也没有输出。
->
尽管事件正在持续发送，我的 DataStream 程序还是没有输出。

File: gettinghelp.zh.md [hunk 8]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
 
+如果你的 DataStream 程序使用了 *事件时间*，那么请检查你的 Watermark 是否已经更新。如果没有产生 Watermark，
+事件时间窗口可能永远不会触发，程序将不会产生任何结果。
 
-### My DataStream application produces no output, even though events are going in.
+你可以在 Flink 的 Web UI（Watermark 部分）中查看 Watermark 是否正在取得进展。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 是否正在取得进展
->
是否正在更新

File: gettinghelp.zh.md [hunk 9]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
-If you run Flink with a very high parallelism, you may need to increase the number of network buffers.
+详情请参考 [配置参考]({{ site.docs-snapshot }}/ops/config.html#configuring-the-network-buffers)。
 
-By default, Flink takes 10% of the JVM heap size for network buffers, with a minimum of 64MB and a maximum of 1GB.
-You can adjust all these values via `taskmanager.network.memory.fraction`, `taskmanager.network.memory.min`, and
-`taskmanager.network.memory.max`.
+### 我的 Job 因为 HDFS/Hadoop 代码的各种异常失败了，我该怎么办？
 
-Please refer to the [Configuration Reference]({{ site.docs-snapshot }}/ops/config.html#configuring-the-network-buffers) for details.
+最常见的原因是 Flink 的类路径中的 Hadoop 版本与你要访问的 Hadoop 群集（HDFS / YARN）版本不同。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: Hadoop 群集
->
Hadoop 集群

File: gettinghelp.zh.md [hunk 10]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
 
-### My job fails with various exceptions from the HDFS/Hadoop code. What can I do?
-
-The most common cause for that is that the Hadoop version in Flink's classpath is different than the
-Hadoop version of the cluster you want to connect to (HDFS / YARN).
-
-The easiest way to fix that is to pick a Hadoop-free Flink version and simply export the Hadoop path and
-classpath from the cluster.
+解决这个问题的最简单方法是选择一个不含 Hadoop 的 Flink 版本，并从集群中导出 Hadoop 路径和类路径即可。
```
Comment (kaibozhou) [inline gettinghelp.zh.md]: 并从集群中导出 Hadoop 路径和类路径即可。
->
并在集群中通过 export 的方式设置 Hadoop 路径和类路径即可。

File: gettinghelp.zh.md [hunk 11]
```diff
@@ -6,128 +6,110 @@ title: "获取帮助"
...
+如果你发现一个意外行为可能是由 Bug 导致的，你可以在 [Flink's JIRA](https://issues.apache.org/jira/issues/?jql=project %3D FLINK) 中搜索已经上报的 Bug 或者发布该 Bug。
 
-If you are unsure whether the unexpected behavior happend due to a bug or not, please post a question to the [user mailing list](#user-mailing-list).
+如果你不确定意外行为的发生是否由 Bug 引起的，请发送问题到 [用户邮件列表](#user-mailing-list)。
 
-## Got an Error Message?
+## 收到错误信息?
 
-Identifying the cause for an error message can be challenging. In the following, we list the most common error messages and explain how to handle them.
+识别错误消息的原因可能具有挑战性。在下文中，我们列出了最常见的错误消息并解释了如何处理它们。
```
[the remark this answers is not in the document]
Comment (sunnyholyd) [reply gettinghelp.zh.md]: "找到导致错误的原因通常是比较困难的。"  这样意译可以吗？

**Comment by sunnyholyd** [comment]
@kaibozhou Thanks for your review. I have pushed a amended commit.

**Comment by kaibozhou** [comment]
Hi @sunnyholyd thanks for the great work. The translation looks good to me.

Best
Kaibo

**Comment by wuchong** [comment]
Thanks for the great work @sunnyholyd  @kaibozhou . It looks good to me too. 

Will merge this. 

Btw, please avoid "merge" commit in the history, use `rebase` instead of `merge` when you want to update to latest master branch (here is asf-site branch) @sunnyholyd . 

https://flink.apache.org/zh/contribute-code.html#section-5

**Comment by sunnyholyd** [comment]
@wuchong Thanks for your reminder, I get it!  : )

**Comment by wuchong** [comment]
Hi @sunnyholyd , I have some troubles when merging your commits. Could you please rebase your branch to latest `asf-site` to avoid the "merge" commits in the history.

**Comment by sunnyholyd** [comment]
Hi @wuchong , I forced to roll back the merge commit. Can you try again?  Thank you.

**Comment by wuchong** [comment]
Thanks @sunnyholyd . I will try again.

**Pull request closed without merging**